File: app/models/fund_breakdown.py
from app.extensions import db


class FundBreakdown(db.Model):
    __tablename__ = "fund_breakdowns"

    id = db.Column(db.Integer, primary_key=True)
    snapshot_id = db.Column(db.Integer, db.ForeignKey("snapshots.id", ondelete="CASCADE"), nullable=False)
    department = db.Column(db.String(50), nullable=False)
    category = db.Column(db.String(50), default="primary")
    fund_name = db.Column(db.String(255), nullable=False)
    amount = db.Column(db.Numeric(12, 2))
    pct_of_total = db.Column(db.Numeric(8, 6))
    # Annual Giving & Direct Mail extra columns
    onetime_count = db.Column(db.Integer)
    recurring_count = db.Column(db.Integer)
    online_count = db.Column(db.Integer)
    mailed_in_count = db.Column(db.Integer)
    total_count = db.Column(db.Integer)

    def __repr__(self):
        return f"<FundBreakdown {self.department}/{self.fund_name}>"

File: src/models/fundBreakdown.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  return sequelize.define('FundBreakdown', {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    snapshotId: { type: DataTypes.INTEGER, allowNull: false, field: 'snapshot_id' },
    department: { type: DataTypes.STRING(50), allowNull: false },
    category: { type: DataTypes.STRING(50), defaultValue: 'primary' },
    fundName: { type: DataTypes.STRING(255), allowNull: false, field: 'fund_name' },
    amount: { type: DataTypes.DECIMAL(12, 2) },
    pctOfTotal: { type: DataTypes.DECIMAL(12, 6), field: 'pct_of_total' },
    onetimeCount: { type: DataTypes.INTEGER, field: 'onetime_count' },
    recurringCount: { type: DataTypes.INTEGER, field: 'recurring_count' },
    onlineCount: { type: DataTypes.INTEGER, field: 'online_count' },
    mailedInCount: { type: DataTypes.INTEGER, field: 'mailed_in_count' },
    totalCount: { type: DataTypes.INTEGER, field: 'total_count' },
  }, {
    tableName: 'fund_breakdowns',
    timestamps: false,
  });
};
